### PR TITLE
Explicitly load Saxon, without incurring any side impacts on the JVM

### LINKED ISSUE #5108: XMLSchemaConverter breaks foreign web applications
XMLSchemaConverter [redefines the JVM system variable `javax.xml.transform.TransformerFactory`](/kitodo/kitodo-production/blob/0c5c76bbb4c2afa9f6b4e2ed9093c4d82854f586/Kitodo-XML-SchemaConverter/src/main/java/org/kitodo/xmlschemaconverter/XMLSchemaConverter.java#L71). This has implications for other web applications running in the same JVM / Tomcat instance, when they want to process XML, they suddenly try to load `net.sf.saxon.TransformerFactoryImpl`, which they don't even have in their classpath, because it is not intended by the developer, and fail with `ClassNotFoundException`.

Workaround: Copy `Saxon-HE-9.9.1-5.jar` from `webapps/kitodo/WEB-INF/lib/` to the `WEB-INF/lib/` directory of the broken web app, then restart Tomcat.

**Goal:** Instead of redefining `javax.xml.transform.TransformerFactory` and using `TransformerFactory.newInstance()`, the code should use `new net.sf.saxon.TransformerFactoryImpl()` to load Saxon explicitly.

### PR DESCRIPTION
**Pull request by matthias-ronge**
Fixes #5108

### CODE REVIEW

**Review by solth — CHANGES_REQUESTED** [review]
Please rebase / resolve conflicts!

**Comment by Kathrin-Huber** [comment]
still waiting for a rebase here

**Comment by matthias-ronge** [comment]
Doesn’t work anymore

File: Kitodo-XML-SchemaConverter/pom.xml [hunk 1]
```diff
@@ -43,6 +43,7 @@
         <dependency>
             <groupId>net.sf.saxon</groupId>
             <artifactId>Saxon-HE</artifactId>
+            <version>${saxon.version}</version>
```
Comment (henning-gerhardt) [inline Kitodo-XML-SchemaConverter/pom.xml:46]: This should not necessary as the used version of a package is controlled in the global pom file.

**Review by solth — APPROVED** [review]
Tested changes with catalog import and it worked as before.

**Pull request merged**